quinn-proto/src/token.rs

```diff
@@ -102,7 +102,7 @@ impl RetryToken {
         let mut sealed_token = raw_token_bytes.to_vec();
 
         let data = aead_key.open(&mut sealed_token, &[])?;
-        let mut reader = io::Cursor::new(data);
+        let mut reader = &mut &data[..];
```

Comment:
I think the `&mut` here is redundant. The variable itself is already mutable, and that's what we take a ref to when calling `decode_addr`.